src/irmin/proof_intf.ml

```diff
@@ -180,13 +195,14 @@ module type Env = sig
   (** {2 Modes} *)
 
   val mode : t -> mode option
-  val to_mode : t -> mode -> unit
-  val with_mode : t -> mode -> (unit -> 'a Lwt.t) -> 'a Lwt.t
+  val to_mode : t -> kind -> mode -> unit
+  val with_mode : t -> kind -> mode -> (unit -> 'a Lwt.t) -> 'a Lwt.t
+  val with_stream : stream -> (t -> 'a Lwt.t) -> 'a Lwt.t
 
   (** {2 In/out backend objects with [Tree]} *)
 
   val add_contents_from_store : t -> hash -> contents -> unit
-  val add_node_from_store : t -> hash -> node -> unit
+  val add_node_from_store : t -> hash -> node -> node
```

Comment:
why return the node?

Reply:
[add_node_from_store] returns a [node] and not [unit] because [Env] may take the opportunity to wrap the input node in [Node.Val.with_handler]

Reply:
because we are modifying the internal `find` function to register the recursive finds.

Reply:
thanks, I somehow missed it